Mathlib/GroupTheory/Perm/Support.lean

```diff
@@ -0,0 +1,670 @@
+    exact ⟨h, hy⟩
+    refine' ⟨hy, heq⟩
+#align
+  equiv.perm.mem_support_swap_mul_imp_mem_support_ne
+  Equiv.Perm.mem_support_swap_mul_imp_mem_support_ne
+
+theorem Disjoint.mem_imp (h : Disjoint f g) {x : α} (hx : x ∈ f.Support) : x ∉ g.Support :=
+  disjoint_left.mp h.disjoint_support hx
+#align equiv.perm.disjoint.mem_imp Equiv.Perm.Disjoint.mem_imp
+
+theorem eq_on_support_mem_disjoint {l : List (Perm α)} (h : f ∈ l) (hl : l.Pairwise Disjoint) :
+    ∀ x ∈ f.Support, f x = l.prod x := by
+  induction' l with hd tl IH
+  · simp at h
+  · intro x hx
+    rw [List.pairwise_cons] at hl
+    rw [List.mem_cons] at h
+    rcases h with (rfl | h)
+    ·
+      rw [List.prod_cons, mul_apply,
+        not_mem_support.mp ((disjoint_prod_right tl hl.left).mem_imp hx)]
+    · rw [List.prod_cons, mul_apply, ← IH h hl.right _ hx, eq_comm, ← not_mem_support]
+      refine' (hl.left _ h).symm.mem_imp _
+      simpa using hx
+#align equiv.perm.eq_on_support_mem_disjoint Equiv.Perm.eq_on_support_mem_disjoint
+
+theorem Disjoint.mono {x y : Perm α} (h : Disjoint f g) (hf : x.Support ≤ f.Support)
+    (hg : y.Support ≤ g.Support) : Disjoint x y := by
+  rw [disjoint_iff_disjoint_support] at h⊢
+  exact h.mono hf hg
+#align equiv.perm.disjoint.mono Equiv.Perm.Disjoint.mono
+
+theorem support_le_prod_of_mem {l : List (Perm α)} (h : f ∈ l) (hl : l.Pairwise Disjoint) :
+    f.Support ≤ l.prod.Support := by
+  intro x hx
+  rwa [mem_support, ← eq_on_support_mem_disjoint h hl _ hx, ← mem_support]
+#align equiv.perm.support_le_prod_of_mem Equiv.Perm.support_le_prod_of_mem
+
+section ExtendDomain
+
+variable {β : Type _} [DecidableEq β] [Fintype β] {p : β → Prop} [DecidablePred p]
+
+@[simp]
+theorem support_extend_domain (f : α ≃ Subtype p) {g : Perm α} :
+    Support (g.extendDomain f) = g.Support.map f.asEmbedding := by
+  ext b
+  simp only [exists_prop, Function.Embedding.coeFn_mk, toEmbedding_apply, mem_map, Ne.def,
+    Function.Embedding.trans_apply, mem_support]
+  by_cases pb : p b
+  · rw [extendDomain_apply_subtype _ _ pb]
+    constructor
+    · rintro h
+      refine' ⟨f.symm ⟨b, pb⟩, _, by simp⟩
+      contrapose! h
+      simp [h]
+    · rintro ⟨a, ha, hb⟩
+      contrapose! ha
+      obtain rfl : a = f.symm ⟨b, pb⟩ := by
+        rw [eq_symm_apply]
+        exact Subtype.coe_injective hb
+      rw [eq_symm_apply]
+      exact Subtype.coe_injective ha
+  · rw [extendDomain_apply_not_subtype _ _ pb]
+    simp only [not_exists, false_iff_iff, not_and, eq_self_iff_true, not_true]
+    rintro a _ rfl
+    exact pb (Subtype.prop _)
+#align equiv.perm.support_extend_domain Equiv.Perm.support_extend_domain
+
+theorem card_support_extend_domain (f : α ≃ Subtype p) {g : Perm α} :
+    (g.extendDomain f).Support.card = g.Support.card := by simp
+#align equiv.perm.card_support_extend_domain Equiv.Perm.card_support_extend_domain
+
+end ExtendDomain
+
+section Card
+
+-- Porting note: lower priority to avoid linter complaints about simp-normal form
+@[simp 1200]
```

Comment:
Here the linter informs that it can be proved with
`by simp only [@card_eq_zero, @Perm.support_eq_empty_iff]`
But none of these seem to be duplicates of this one.